probe_src/python/probe_py/manual/cli.py

```diff
@@ -169,5 +172,138 @@ def dump(
                     print(pid, exid, tid, op_no, op.data)
                 print()
 
+@app.command(
+    context_settings=dict(
+        ignore_unknown_options=True,
+    ),
+)
+def ssh(
+        ssh_args: list[str],
+        debug: bool = typer.Option(default=False, help="Run verbose & debug build of libprobe"),
+):
+    """
+    Wrap SSH and record provenance of the remote command.
+    """
+
+    one_arg_options = set("BbcDEeFIiJLlmOoPpRSWw")
+    no_arg_options = set("46AaCfGgKkMNnqsTtVvXxYy")
+
+    # fsm to figure out the flags, destination and remote cmds
+    state = 'start'
+    i = 0
+    flags = []
+    destination = None
+    remote_host = []
+
+    while i < len(ssh_args):
```

Comment:
This looks great!

Let's pull it out into a new function, `ssh_arg_parse(args: list[str]) ->tuple[list[str], str, list[str]]` (first elem is the SSH args, then the hostname, then the cmd) in its own module `ssh.py`.

Then you should write a few test-cases in a function called `test_ssh_arg_parse`.